Sources/Shared/Data Manager/Proxies/Social/MobileFileUploadManager.m

```diff
@@ -0,0 +1,269 @@
+    
+    // Upload File process
+    // 1. Check if destination folder exists on the server
+    // 2. Upload file with ECMS upload service (method POST)
+    
+    NSString * mobileFolderPath = [self mobileFolderPath:space];
+    NSMutableURLRequest* mobileFolderCheckRequest = [[NSMutableURLRequest alloc] init];
+    [mobileFolderCheckRequest setURL:[NSURL URLWithString:mobileFolderPath]];
+    [mobileFolderCheckRequest setHTTPMethod:@"HEAD"];
+    [mobileFolderCheckRequest setValue:kUserAgentHeader forHTTPHeaderField:@"User-Agent"];
+    
+    NSURLSessionConfiguration * config = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession * mobileFolderCheckSession = [NSURLSession sessionWithConfiguration:config delegate:nil delegateQueue:nil];
+    
+    NSURLSessionDataTask *checkMobileFolderTask = [mobileFolderCheckSession dataTaskWithRequest:mobileFolderCheckRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (!error) {
+            NSUInteger statusCode = [((NSHTTPURLResponse*) response) statusCode];
+            if (statusCode >= 200 && statusCode < 300) {
+                // Folder exists: start the upload process
+                [self startUploadFile:fileName data:fileData toSpace:space progess:progressHandler completionHandler:completion];
+            } else {
+                // Mobile folder doesn't exist: send a request (method:MKCOL) to create the mobile folder
+                NSMutableURLRequest* createMobileFolderRequest = [[NSMutableURLRequest alloc] init];
+                [createMobileFolderRequest setURL:[NSURL URLWithString:mobileFolderPath]];
+                [createMobileFolderRequest setHTTPMethod:@"MKCOL"];
+                [createMobileFolderRequest setValue:kUserAgentHeader forHTTPHeaderField:@"User-Agent"];
+                
+                NSURLSessionDataTask *createMobileFolderTask = [mobileFolderCheckSession dataTaskWithRequest:createMobileFolderRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                    NSUInteger statusCode = [((NSHTTPURLResponse*) response) statusCode];
+                    if(statusCode >= 200 && statusCode < 300) {
+                        // Folder created: start the upload process
+                        [self startUploadFile:fileName data:fileData toSpace:space progess:progressHandler completionHandler:completion];
+                    } else {
+                        completion (nil, NO, error);
+                    }
+                }];
+                // Start the task to create the Mobile folder
+                [createMobileFolderTask resume];
+            }
+        }
+    }];
+    // Start the task to check if the Mobile folder exists
+    [checkMobileFolderTask resume];
+}
+
+-(void) moveUploadedFileToDestination:(SocialSpace*)space fileName:(NSString*)fileAttachName uploadId:(NSString*)uploadId completionHandler:(void (^)(NSString *, BOOL, NSError *))completion {
```

Comment:
Created this method to separate these actions from the `startUploadFile` method which was very big